### PR TITLE
Take care not to dereference NULL element

### PR DESCRIPTION
**Pull request by janisozaur**

### CODE REVIEW

File: src/world/map.c [hunk 1]
```diff
@@ -750,6 +750,10 @@ void game_command_remove_scenery(int* eax, int* ebx, int* ecx, int* edx, int* es
 
 	bool sceneryFound = false;
 	rct_map_element* map_element = map_get_first_element_at(x / 32, y / 32);
```
Comment (IntelOrca) [inline src/world/map.c]: We assume every tile to have a surface element, and even create one if it doesn't exist when we load a park. I don't think there is any point doing a check here and no where else... unless of course the x and y position is out of bounds. In which case that is a problem for most game commands.

Comment (janisozaur) [inline src/world/map.c]: OK, I'll change it to validate `x` and `y` values earlier.

File: src/world/map.c [hunk 2]
```diff
@@ -2650,10 +2654,16 @@ void game_command_remove_fence(int* eax, int* ebx, int* ecx, int* edx, int* esi,
...
-	while(map_element_get_type(map_element) != MAP_ELEMENT_TYPE_FENCE ||
-	map_element->base_height != base_height ||
-	(map_element->type & MAP_ELEMENT_DIRECTION_MASK) != direction ||
-	((*ebx & 0x40) && !(map_element->flags & MAP_ELEMENT_FLAG_GHOST))){
+	if (map_element == NULL) {
+		*ebx = MONEY32_UNDEFINED;
+		return;
+	}
+
+	while (map_element_get_type(map_element) != MAP_ELEMENT_TYPE_FENCE ||
```
Comment (duncanspumpkin) [inline src/world/map.c]: This loop needs to be rewritten in the style of the rest of the code base.

**Comment by janisozaur** [comment]
updated

**Pull request merged**